grails-app/services/org/pih/warehouse/inventory/StocklistService.groovy

```diff
@@ -83,4 +91,140 @@ class StocklistService {
         }
         mailService.sendHtmlMailWithAttachment(to, subject, body, attachments)
     }
+
+    void clearStockListItems(String id) {
+        Requisition requisition = Requisition.get(id)
```

Comment:
Move this part to the stocklist controller and instead of throwing ObjectNotFoundException, you should return 404 status.